### ISSUE TITLE
escape sequences in string literals passed to extern functions break CUDA code generation

### ISSUE THREAD

**Issue by michael-kenzel** [AnyDSL/impala#85]
example: we add
```rust
extern "device" {
	fn "printf" cuda_device_print_i32(&[u8], i32) -> i32;
}
```
and call
```rs
cuda_device_print_i32("blub %d\n", 42);
```
the generated CUDA C++ code will end up looking like
```cpp
        printf_99098 = printf("blub %d
", 42);
```
resulting in compilation failure.

The problem would seem to be that CUDA C++ code generation does not escape string contents before outputting string literals. "pre-escaping" the string in the original impala source code would seem to be a temporary workaround:
```rs
cuda_device_print_i32("blub %d\\n", 42);
```

**Comment by leissa** [comment]
@richardmembarth can you have a look at this?

**Comment by richardmembarth** [comment]
Fixed in AnyDSL/thorin@d93d0def
